testing/opensmtpd-extras/APKBUILD

```diff
@@ -0,0 +1,134 @@
+# Contributor: Shiz <hi@shiz.me>
+# Maintainer: Shiz <hi@shiz.me>
+pkgname=opensmtpd-extras
+pkgver=201703132115
+pkgrel=0
+pkgdesc="OpenSMTPD addons"
+url="https://opensmtpd.org/"
+arch="all"
+license="ISC"
+depends="libevent"
+depends_filter_monkey="libressl"
+depends_queue_python="python2"
+depends_scheduler_python="python2"
+depends_table_mysql="mariadb-client-libs"
+depends_table_postgres="libpq"
+depends_table_python="python2"
+depends_table_redis="hiredis"
+depends_table_sqlite="sqlite-libs"
+makedepends="libressl-dev libevent-dev"
+makedepends_queue_python="python2-dev"
+makedepends_scheduler_python="python2-dev"
+makedepends_table_mysql="mariadb-dev"
+makedepends_table_postgres="postgresql-dev"
+makedepends_table_python="python2"
+makedepends_table_redis="hiredis-dev"
+makedepends_table_sqlite="sqlite-dev"
+subpackages=""
+source="https://www.opensmtpd.org/archives/opensmtpd-extras-$pkgver.tar.gz
+	remove-decls.patch"
+builddir="$srcdir/opensmtpd-extras-$pkgver"
+
+_extras="
+	filter-monkey
+	filter-stub
+	filter-trace
+	filter-void
+	queue-null
+	queue-python
+	queue-ram
+	queue-stub
+	scheduler-python
+	scheduler-ram
+	scheduler-stub
+	table-ldap
+	table-mysql
+	table-passwd
+	table-postgres
+	table-python
+	table-redis
+	table-socketmap
+	table-sqlite
+	table-stub
+"
+_extradocs="
+	filter-monkey
+	filter-stub
+	filter-trace
+	filter-void
+	table-sqlite
+	table-redis
+	table-socketmap
+	table-passwd
+"
+
+for _extra in $_extras; do
+	_extraident="${_extra/-/_}"
+	_extramakedeps="$(eval "echo \$makedepends_$_extraident")"
+	subpackages="$subpackages opensmtpd-$_extra:_package_extra"
+	makedepends="$makedepends $_extramakedeps"
+	case "$_extradocs" in *$_extra*) subpackages="$subpackages opensmtpd-$_extra-doc:_package_extra_doc" ;; esac
+done
+subpackages="$subpackages $pkgname-doc"
+
+build() {
+	cd "$builddir"
+	local with_extras
+	local extra; for extra in $_extras; do
+		with_extras="$with_extras --with-$extra"
+	done
+
+	./configure \
+		--prefix=/usr \
+		--sysconfdir=/etc/smtpd \
+		--with-libssl=/usr/include/openssl \
+		--mandir=/usr/share/man \
+		--libexecdir=/usr/lib \
+		--with-pie \
+		--with-mantype=man \
+		$with_extras \
+		|| return 1
+	# the Redis table expects <hiredis.h> without subdirectory, so indulge it
+	make REDIS_CPPFLAGS='-I /usr/include/hiredis' || return 1
+}
+
+package() {
+	cd "$builddir"
+	make DESTDIR="$pkgdir" install || return 1
+}
+
+_package_extra() {
+	local extraname; extraname="${subpkgname#opensmtpd-}"
```

Comment:
`local` itself doesn’t work in all shells, it’s not defined by POSIX. However, all common shells (ash, dash, ZSH, Bash) support it and none of them has problem with `local foo=bar`. I don’t know about any shell that supports `local`, but not with assignment.